### PR TITLE
fix(components): [input-number] correct input event behavior

### LINKED ISSUE #9841: [Component] [input-number] el-input-number component validate on blur even set trigger to change
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.38`
- Element Plus Version: `2.2.17`
- Browser / OS: `Chrome 105 / Windows 10`
- Build Tool: `Vite`

### Related Component

- `el-input-number`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZm9ybSBjbGFzcz1cImZvcm1cIiByZWY9XCJmb3JtUmVmXCIgOm1vZGVsPVwiZm9ybVN0YXRlXCIgc2l6ZT1cImxhcmdlXCI+XG4gICAgPHA+XG4gICAgICBXaGVuIHNldCBgdHJpZ2dlcmAgdG8gYGNoYW5nZWAgaW4gYHJ1bGVzYCwgdGhlIGBlbC1pbnB1dC1udW1iZXJgIGNvbXBvbmVudFxuICAgICAgc3RpbGwgdmFsaWRhdGUgd2hlbiBgYmx1cmAuXG4gICAgPC9wPlxuICAgIDxlbC1mb3JtLWl0ZW1cbiAgICAgIGxhYmVsPVwibnVtYmVyXCJcbiAgICAgIDpydWxlcz1cIltcbiAgICAgICAge1xuICAgICAgICAgIHJlcXVpcmVkOiB0cnVlLFxuICAgICAgICAgIHRyaWdnZXI6ICdjaGFuZ2UnLFxuICAgICAgICAgIG1lc3NhZ2U6ICdudW1iZXIgc2hvdWxkIG5vdCBiZSBlbXB0eScsXG4gICAgICAgIH0sXG4gICAgICBdXCJcbiAgICAgIHByb3A9XCJudW1iZXJcIlxuICAgID5cbiAgICAgIDxlbC1pbnB1dC1udW1iZXJcbiAgICAgICAgOm1pbj1cIjEwXCJcbiAgICAgICAgOm1heD1cIjEwMFwiXG4gICAgICAgIHYtbW9kZWw9XCJmb3JtU3RhdGUubnVtYmVyXCJcbiAgICAgID48L2VsLWlucHV0LW51bWJlcj5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgICA8cD5gZWwtaW5wdXRgIGNvbXBvbmVudCBpcyBub3JtYWwuPC9wPlxuICAgIDxlbC1mb3JtLWl0ZW1cbiAgICAgIGxhYmVsPVwidGV4dFwiXG4gICAgICA6cnVsZXM9XCJbXG4gICAgICAgIHtcbiAgICAgICAgICByZXF1aXJlZDogdHJ1ZSxcbiAgICAgICAgICB0cmlnZ2VyOiAnY2hhbmdlJyxcbiAgICAgICAgICBtZXNzYWdlOiAndGV4dCBzaG91bGQgbm90IGJlIGVtcHR5JyxcbiAgICAgICAgfSxcbiAgICAgIF1cIlxuICAgICAgcHJvcD1cInRleHRcIlxuICAgID5cbiAgICAgIDxlbC1pbnB1dCB2LW1vZGVsPVwiZm9ybVN0YXRlLnRleHRcIj48L2VsLWlucHV0PlxuICAgIDwvZWwtZm9ybS1pdGVtPlxuICA8L2VsLWZvcm0+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwIGxhbmc9XCJ0c1wiPlxuaW1wb3J0IHsgcmVhY3RpdmUsIHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB0eXBlIHsgRm9ybUluc3RhbmNlIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5jb25zdCBmb3JtUmVmID0gcmVmPEZvcm1JbnN0YW5jZT4oKVxuY29uc3QgZm9ybVN0YXRlID0gcmVhY3RpdmU8eyBudW1iZXI/OiBudW1iZXI7IHRleHQ/OiBzdHJpbmcgfT4oe30pXG48L3NjcmlwdD5cblxuPHN0eWxlPlxuYm9keSB7XG4gIG1hcmdpbjogMDtcbiAgaGVpZ2h0OiAxMDB2aDtcbiAgZGlzcGxheTogZmxleDtcbiAganVzdGlmeS1jb250ZW50OiBjZW50ZXI7XG4gIGFsaWduLWl0ZW1zOiBjZW50ZXI7XG4gIGZvbnQtc2l6ZTogMjRweDtcbiAgYmFja2dyb3VuZC1jb2xvcjogI2VlZTtcbn1cbi5mb3JtIHtcbiAgd2lkdGg6IDcwMHB4O1xuICBiYWNrZ3JvdW5kLWNvbG9yOiB3aGl0ZTtcbiAgcGFkZGluZzogMjBweDtcbn1cbjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

Set `trigger` to `change` in `rules` on `<el-form-item>` component which contains a `<el-input-number>` component .

### What is Expected?

Validate on change

### What is actually happening?

Validate on blur

### PR DESCRIPTION
**Pull request by holazz**
Consistent with native numeric input behavior, closes #9841

[Before](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtaW5wdXQtbnVtYmVyIHYtbW9kZWw9XCJ2YWx1ZVwiIEBpbnB1dD1cImhhbmRsZUlucHV0XCIgLz5cbiAgPGlucHV0IHR5cGU9XCJudW1iZXJcIiB2LW1vZGVsPVwidmFsdWUxXCIgQGlucHV0PVwiaGFuZGxlSW5wdXQxXCIgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCB2YWx1ZSA9IHJlZigxKVxuY29uc3QgdmFsdWUxID0gcmVmKDEpXG5cbmNvbnN0IGhhbmRsZUlucHV0ID0gKCkgPT4gY29uc29sZS5sb2coJ2VsLWlucHV0JylcbmNvbnN0IGhhbmRsZUlucHV0MSA9ICgpID0+IGNvbnNvbGUubG9nKCduYXRpdmUgaW5wdXQnKVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

[After](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtaW5wdXQtbnVtYmVyIHYtbW9kZWw9XCJ2YWx1ZVwiIEBpbnB1dD1cImhhbmRsZUlucHV0XCIgLz5cbiAgPGlucHV0IHR5cGU9XCJudW1iZXJcIiB2LW1vZGVsPVwidmFsdWUxXCIgQGlucHV0PVwiaGFuZGxlSW5wdXQxXCIgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCB2YWx1ZSA9IHJlZigxKVxuY29uc3QgdmFsdWUxID0gcmVmKDEpXG5cbmNvbnN0IGhhbmRsZUlucHV0ID0gKCkgPT4gY29uc29sZS5sb2coJ2VsLWlucHV0JylcbmNvbnN0IGhhbmRsZUlucHV0MSA9ICgpID0+IGNvbnNvbGUubG9nKCduYXRpdmUgaW5wdXQnKVxuPC9zY3JpcHQ+XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctOTg1MC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTk4NTAtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgIGxpbmsuaHJlZiA9ICdodHRwczovL3ByZXZpZXctOTg1MC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy05ODUwLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -179,12 +185,14 @@ const increase = () => {
   const value = props.modelValue || 0
   const newVal = ensurePrecision(value)
   setCurrentValue(newVal)
+  emit(INPUT_EVENT, data.currentValue)
```
Comment (sxzz) [inline packages/components/input-number/src/input-number.vue:188]: It seems will be emitted in `setCurrentValue`. Why to repeat it?

File: packages/components/input-number/src/input-number.vue [hunk 2]
```diff
@@ -179,12 +185,14 @@ const increase = () => {
...
 }
 const decrease = () => {
   if (props.readonly || inputNumberDisabled.value || minDisabled.value) return
   const value = props.modelValue || 0
   const newVal = ensurePrecision(value, -1)
   setCurrentValue(newVal)
+  emit(INPUT_EVENT, data.currentValue)
```
Comment (sxzz) [inline packages/components/input-number/src/input-number.vue:195]: Same here

File: packages/components/input-number/src/input-number.vue [hunk 3]
```diff
@@ -218,16 +226,16 @@ const setCurrentValue = (value: number | string | null | undefined) => {
   const newVal = verifyValue(value)
   if (oldVal === newVal) return
   data.userInput = null
-  emit('update:modelValue', newVal!)
-  emit('input', newVal)
```
Comment (sxzz) [inline packages/components/input-number/src/input-number.vue:222]: It may causes breaking change if we remove `input` event

Comment (holazz) [reply packages/components/input-number/src/input-number.vue:222]: This results in redundant input event being fired.

https://user-images.githubusercontent.com/23100055/191425549-d77f45df-4291-4a86-a019-1634bdd5295d.mov

**Review by sxzz — APPROVED** [review]

**Pull request merged**